sycl/test/basic_tests/image/image_write.cpp

```diff
@@ -0,0 +1,276 @@
+bool test1d_coord(dataT *hostPtr, coordT coord, dataT colour) {
+  dataT resultData;
+
+  s::default_selector selector;
+
+  { // Scope everything to force destruction
+    s::image<1> image(hostPtr, s::image_channel_order::rgba,
+                      channelType, s::range<1>{3});
+
+    s::buffer<dataT, 1> resultDataBuf(&resultData,
+                                      s::range<1>(1));
+
+    // Do the test by reading a single pixel from the image
+    s::queue myQueue(selector);
+
+    myQueue.submit([&](s::handler &cgh) {
+      auto imageAcc = image.get_access<dataT, s::access::mode::write>(cgh);
+      cgh.single_task<test_1d_write_class>([=]() {
+        imageAcc.write(coord, colour);
+      });
+    });
+
+    myQueue.submit([&](s::handler &cgh) {
+      auto imageAcc = image.get_access<dataT, s::access::mode::read>(cgh);
+      s::accessor<dataT, 1, s::access::mode::write>
+          resultDataAcc(resultDataBuf, cgh);
+
+      cgh.single_task<test_1d_read_class>([=]() {
+        dataT RetColor = imageAcc.read(coord);
+        resultDataAcc[0] = RetColor;
+      });
+    });
+  }
+#ifdef PRINT_RESULT
+  std::cout << "Expected: " << colour.r() << ", " << colour.g()
+            << ", " << colour.b() << ", " << colour.a() << "\n";
+  std::cout << "Got:      " << resultData.r() << ", " << resultData.g()
+            << ", " << resultData.b() << ", " << resultData.a() << "\n";
+#endif // PRINT_RESULT
+  bool correct = true;
+  if (resultData.r() != colour.r())
+    correct = false;
+  if (resultData.g() != colour.g())
+    correct = false;
+  if (resultData.b() != colour.b())
+    correct = false;
+  if (resultData.a() != colour.a())
+    correct = false;
+  return correct;
+}
+
+template <typename dataT, typename coordT, s::image_channel_type channelType>
+bool test2d_coord(dataT *hostPtr, coordT coord, dataT colour) {
+  dataT resultData;
+
+  s::default_selector selector;
+
+  { // Scope everything to force destruction
+    s::image<2> image(hostPtr, s::image_channel_order::rgba,
+                      channelType, s::range<2>{3, 3});
+
+    s::buffer<dataT, 1> resultDataBuf(&resultData,
+                                      s::range<1>(1));
+
+    // Do the test by reading a single pixel from the image
+    s::queue myQueue(selector);
+
+    myQueue.submit([&](s::handler &cgh) {
+      auto imageAcc = image.get_access<dataT, s::access::mode::write>(cgh);
+
+      cgh.single_task<test_2d_write_class>([=]() {
+        imageAcc.write(coord, colour);
+      });
+    });
+
+    myQueue.submit([&](s::handler &cgh) {
+      auto imageAcc = image.get_access<dataT, s::access::mode::read>(cgh);
+      s::accessor<dataT, 1, s::access::mode::write>
+          resultDataAcc(resultDataBuf, cgh);
+
+      cgh.single_task<test_2d_read_class>([=]() {
+        dataT RetColor = imageAcc.read(coord);
+        resultDataAcc[0] = RetColor;
+      });
+    });
+  }
+#ifdef PRINT_RESULT
```

Comment:
do we need this code as soon as we do not pass this definition to compiler?

Reply:
This can be removed. I will update the PR.